File: Exercise_1/Exercise_1.cpp
#include <iostream>
#include <vector>

bool isPrime(unsigned int n) {
  for (unsigned int i = 2; i < n; i++) {
    if (n%i == 0) {
      return false;
    }
  }
  return true;
}


int main() {

  unsigned int max = 1000000;
  std::vector<unsigned int> primes;
  for (unsigned int i = 2; i < max; ++i) {
    if (isPrime(i)) {
      primes.push_back(i); 
      //std::cout << i << std::endl;
    }
  }
  //for (unsigned int prime : primes) {
  //  std::cout << prime << std::endl;
  //}
  std::cout << "There are " << primes.size() << "primes below " << max << std::endl;

  return 0;
}

File: Exercise_1/Exercise_1.py



# Find total number of primes below 1 million
# time python ./Exercise_1.py

def isPrime(n) :
  for i in xrange(2,n):
    if (n%i == 0):
      return False

  return True

def main():

  maxp = 100000
  primes = []
  for i in xrange(2,maxp):
    if isPrime(i) :
      primes.append(i)
  print "There are ",  len(primes), " primes below ", maxp


#Python idiom to get a main function
if __name__ == "__main__":
  main()

